util/secret/certificates.go

```diff
@@ -215,10 +232,11 @@ func (c Certificates) Generate() error {
 				continue
 			case ServiceAccount:
 				generator = generateServiceAccountKeys
+			case ClusterAPIEtcdClient: // Depends on the etcd CA, so needs to be handled separately
```

Comment:
I found the `Generate` hard to understand; good candidate for a refactor in the future, I think.

Reply:
Yeah, it's painful/very organic.